commit 19a4346601d09963ec646dfc89e640226128a0b9
Author: Matthew Glazar <strager.nds@gmail.com>
Date:   Sun Mar 10 12:29:35 2019 -0700

    @untested Allow configuring WATCHMAN_CONFIG_FILE at build time

    Watchman's old autoconf-based build system allowed customizing the value
    of WATCHMAN_CONFIG_FILE. Watchman's new CMake-based build system does
    not. Allow users to configure WATCHMAN_CONFIG_FILE when using the CMake
    build system.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb282818..e70c84d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,6 +92,15 @@ include(CheckIncludeFiles)
 include(CheckStructHasMember)
 include(CheckSymbolExists)

+if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+  set(WATCHMAN_DEFAULT_CONFIG_FILE C:/ProgramData/facebook/watchman.json)
+else()
+  set(WATCHMAN_DEFAULT_CONFIG_FILE /etc/watchman.json)
+endif()
+set(
+  WATCHMAN_CONFIG_FILE "${WATCHMAN_DEFAULT_CONFIG_FILE}"
+  CACHE STRING "Run-time path of the global watchman.json configuration file")
+
 # configure_file wants us to define a separate file.  I'd rather not
 # have boilerplate for the same thing in two difference files, so we
 # roll the checks in together with writing out the features to config.h
@@ -110,11 +119,8 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
 endif()

 config_h("// Generated by cmake")
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-  config_h("#define WATCHMAN_CONFIG_FILE \
-\"C:/ProgramData/facebook/watchman.json\"")
-else()
-  config_h("#define WATCHMAN_CONFIG_FILE \"/etc/watchman.json\"")
+config_h("#define WATCHMAN_CONFIG_FILE \"${WATCHMAN_CONFIG_FILE}\"")
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
   config_h("#define WATCHMAN_STATE_DIR \"${WATCHMAN_STATE_DIR}\"")
 endif()
 config_h("#define PACKAGE_VERSION \"${PACKAGE_VERSION}\"")
diff --git a/website/_docs/configuration.markdown b/website/_docs/configuration.markdown
index 5ab8381a..e4a8c968 100644
--- a/website/_docs/configuration.markdown
+++ b/website/_docs/configuration.markdown
@@ -16,7 +16,7 @@ in the root directory, watchman will load it and use it as a source of
 configuration information specific to that root.

 The global configuration path can be changed by passing the
-`--enable-conffile` option to configure when you build watchman.  This
+`-DWATCHMAN_CONFIG_FILE` option to CMake when you build watchman.  This
 documentation refers to it as `/etc/watchman.json` throughout, just be aware
 that your particular installation may locate it elsewhere.   In addition,
 the environmental variable `$WATCHMAN_CONFIG_FILE` will override the
